### PR TITLE
Init model and codebase

### PR DESCRIPTION
**Pull request by vntdinh97**

### CODE REVIEW

File: GiaPhaOnline.Domain/Entities/DongHo.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using GiaPhaOnline.Domain.Enums;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GiaPhaOnline.Domain.Entities
+{
+    public class DongHo
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/DongHo.cs]: should be in English, name it 'Family' instead

File: GiaPhaOnline.Domain/Entities/DongHo.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        public int Id { get; set; }
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/DongHo.cs]: use Ulid to create an Id and store it as a string in the database

File: GiaPhaOnline.Domain/Entities/DongHo.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        public string Name { get; set; }
+        public int Address{ get; set; }
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/DongHo.cs]: address should be string

File: GiaPhaOnline.Domain/Enums/Status.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GiaPhaOnline.Domain.Enums
+{
+    public enum Status
+    {
+        Active,
+        InActicve
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Enums/Status.cs]: typo, InActive

File: GiaPhaOnline.Domain/Entities/DongHo.cs (hunk 1 above)
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/DongHo.cs]: should set the class sealed

File: GiaPhaOnline.Domain/Entities/DongHo.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        public string Description { get; set; }
+        public DateTime Dob { get; set; }
+        public Status Status { get; set; }  
+        public int ModifyBy { get; set; }
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/DongHo.cs:18]: not actually need

File: GiaPhaOnline.Domain/Entities/DongHo.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace GiaPhaOnline.Domain.Entities
+{
+    public class DongHo
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public int Address{ get; set; }
+        public string Description { get; set; }
+        public DateTime Dob { get; set; }
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/DongHo.cs]: not actually need

File: GiaPhaOnline.Domain/Entities/DongHo.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        public Status Status { get; set; }  
+        public int ModifyBy { get; set; }
+        public DateTime ModifyDate { get; set; }
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/DongHo.cs]: remove ModifyBy, ModifyDate, consider to apply EventSourcing later

File: GiaPhaOnline.Domain/Entities/DongHo.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        public DateTime CreateDate { get; set; }
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/DongHo.cs]: use DateTimeOffset, change property to CreatedDate

File: GiaPhaOnline.Domain/Entities/UserInfo.cs [hunk 9]
```diff
@@ -0,0 +1,30 @@
+﻿using GiaPhaOnline.Domain.Enums;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GiaPhaOnline.Domain.Entities
+{
+    public class UserInfo
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/UserInfo.cs]: sealed class

File: GiaPhaOnline.Domain/Entities/UserInfo.cs [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        public int Id { get; set; }
+        public string LastName { get; set; }
+        public string FullName { get; set; }
+        public DateTime Dob { get; set; }
+        public string Email { get; set; }
+        public string PhoneNumber { get; set; }
+        public int DongHoId { get; set; }
+        public string Address { get; set; }
+        public int Dad_ID { get; set; }
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/UserInfo.cs]: use PascalCase

File: GiaPhaOnline.Domain/Entities/UserInfo.cs [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        public int Mother_ID { get; }
+        public int VC_ID { get; }
+        public Status Status { get; set; }
+        public DateTime ModifyDate { get; set; }
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/UserInfo.cs]: CreatedDate, ModifiedDate, CreatedBy, ModifiedBy

File: GiaPhaOnline.Domain/Enums/TransactionStatus.cs [hunk 12]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GiaPhaOnline.Domain.Enums
+{
+    public enum TransactionStatus
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Enums/TransactionStatus.cs]: unused

File: GiaPhaOnline.Domain/Configurations/DongHoConfiguration.cs [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GiaPhaOnline.Domain.Configurations
+{
+    public class DongHoConfiguration : IEntityTypeConfiguration<DongHo>
+    {
+        public void Configure(EntityTypeBuilder<DongHo> builder)
+        {
+            builder.ToTable("DongHos");
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Configurations/DongHoConfiguration.cs]: Family

File: GiaPhaOnline.Domain/Entities/UserInfo.cs (hunk 9 above)
Comment (vntdinh97) [reply GiaPhaOnline.Domain/Entities/UserInfo.cs]: class name should be 'Person'

File: GiaPhaOnline.Domain/Entities/DongHo.cs [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+        public int CreateBy { get; set; }
+
+        public List<UserInfo> UserInfos { get; set; }
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/Entities/DongHo.cs]: virtual

File: GiaPhaOnline.Domain/EF/GiaPhaDbContext.cs [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GiaPhaOnline.Domain.EF
+{
+    public class GiaPhaDbContext : DbContext
+    {
+        public GiaPhaDbContext(DbContextOptions options) : base(options)
+        {
```
Comment (vntdinh97) [inline GiaPhaOnline.Domain/EF/GiaPhaDbContext.cs]: Database.EnsureCreated()

File: GiaPhaOnline.Domain/appsettings.json
Comment (vntdinh97) [inline GiaPhaOnline.Domain/appsettings.json:1]: env variable should be obtianed from start project settings, normally Api project, all other projects are added to the Api project as reference projects

File: GiaPhaOnline.Domain/Entities/UserInfo.cs (hunk 9 above)
Comment (QuangDai97) [reply GiaPhaOnline.Domain/Entities/UserInfo.cs]: > sealed class

Cái này chỉ đc set Public với Internal, mà cái internal thì ở config nó báo lỗi pham vi, có cách nào khác không bồ?

File: GiaPhaOnline.Domain/appsettings.json
Comment (QuangDai97) [reply GiaPhaOnline.Domain/appsettings.json:1]: Chỗ này CẤN

**Pull request merged**